packages/wrangler/src/pipelines/index.ts

```diff
@@ -0,0 +1,459 @@
+
+				const pipelineConfig: PipelineUserConfig = {
+					name: name,
+					metadata: {},
+					source: [
+						{
+							type: "http",
+							format: "json",
+						},
+						{
+							type: "binding",
+							format: "json",
+						},
+					],
+					transforms: [],
+					destination: {
+						type: "r2",
+						format: "json",
+						compression: {
+							type: compression,
+						},
+						batch: batch,
+						path: {
+							bucket: bucket,
+						},
+						credentials: {
+							endpoint: getAccountR2Endpoint(accountId),
+							access_key_id: args["access-key-id"] || "",
+							secret_access_key: args["secret-access-key"] || "",
+						},
+					},
+				};
+				const destination = pipelineConfig.destination;
+				if (
+					!destination.credentials.access_key_id &&
+					!destination.credentials.secret_access_key
+				) {
+					// auto-generate a service token
+					await authorizeR2Bucket(name, accountId, pipelineConfig.destination);
+				}
+
+				if (!destination.credentials.access_key_id) {
+					throw new FatalError("Requires a r2 access key id");
+				}
+
+				if (!destination.credentials.secret_access_key) {
+					throw new FatalError("Requires a r2 secret access key");
+				}
+
+				if (args.authentication !== undefined && args.authentication === true) {
+					pipelineConfig.source = [
+						{
+							type: "binding",
+							format: "json",
+						},
+					];
+				}
+
+				if (args.transform !== undefined) {
+					const split = args.transform.split(".");
+					if (split.length === 2) {
+						pipelineConfig.transforms.push({
+							script: split[0],
+							entrypoint: split[1],
+						});
+					} else if (split.length === 1) {
+						pipelineConfig.transforms.push({
+							script: split[0],
+							entrypoint: "Transform",
+						});
+					} else {
+						throw new Error(
+							"invalid transform: required syntax script.entrypoint"
+						);
+					}
+				}
+
+				if (args.filepath) {
+					pipelineConfig.destination.path.filepath = args.filepath;
+				}
+				if (args.filename) {
+					pipelineConfig.destination.path.filename = args.filename;
+				}
+
+				logger.log(`🌀 Creating pipeline named "${name}"`);
+				const pipeline = await createPipeline(accountId, pipelineConfig);
+				await metrics.sendMetricsEvent("create pipeline", {
+					sendMetrics: config.send_metrics,
+				});
+
+				logger.log(
+					`✅ Successfully created pipeline "${pipeline.name}" with ID ${pipeline.id}`
+				);
+				logger.log("🎉 You can now send data to your pipeline!");
+				logger.log(
+					`Example: curl "${pipeline.endpoint}" -d '[{"foo": "bar"}]'`
+				);
+			}
+		)
+		.command(
+			"list",
+			"List current pipelines",
+			(yargs) => yargs,
+			async (args) => {
+				const config = readConfig(args.config, args);
+				const accountId = await requireAuth(config);
+
+				// TODO: we should show bindings & transforms if they exist for given ids
+				const pipelines = await listPipelines(accountId);
```

Comment:
As above, we should handle the situation where remote calls fail.

Reply:
should only fail if the account was invalid, which will give an API error that should have the corresponding error message.